app/models/log_ticket.rb

```diff
@@ -0,0 +1,27 @@
+class LogTicket < ActiveRecord::Base
+  enum backend: [ :s3, :local ]
+
+  scope :latest_pending, -> { limit(1).lock(true).select("id").where(status: "pending") }
+
+  def self.pop(key = nil, directory = nil)
+    scope = latest_pending
+    scope = scope.where(key: key) if key
+    scope = scope.where(directory: directory) if directory
+    sql = scope.to_sql
+
+    find_by_sql(["UPDATE #{quoted_table_name} SET status = ? WHERE id IN (#{sql}) RETURNING *", 'processing']).first
+  end
+
+  def filesystem
+    @fs ||=
+      if s3?
```

Comment:
I agree with moving this into `RubygemsFs.new`, it's pretty brittle to make LogTicket know about the different classes and arguments.